docs/references/amp-constants-example.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (C) 2022 Alteryx, Inc. All rights reserved.
+#
+# Licensed under the ALTERYX SDK AND API LICENSE AGREEMENT;
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.alteryx.com/alteryx-sdk-and-api-license-agreement
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Example input tool."""
+from collections import namedtuple
+
+from ayx_python_sdk.core import PluginV2
+from ayx_python_sdk.providers.amp_provider.amp_provider_v2 import AMPProviderV2
+
+from pyarrow import Table
+
+
+class ConstantsExample(PluginV2):
+    """Concrete implementation of an AyxPlugin."""
+
+    def __init__(self, provider: AMPProviderV2) -> None:
+        """Construct a plugin."""
+        self.provider = provider
+        self.config_value = 0.42
+        self.provider.io.info("Plugin initialized.")
+
+    def on_incoming_connection_complete(self, anchor: namedtuple) -> None:
+        """
+        Call when an incoming connection is done sending data including when no data is sent on an optional input anchor.
+
+        This method IS NOT called during update-only mode.
+
+        Parameters
+        ----------
+        anchor
+            NamedTuple containing anchor.name and anchor.connection.
+        """
+        raise NotImplementedError("Input tools don't receive incoming connections.")
+
+    def on_record_batch(self, batch: "Table", anchor: namedtuple) -> None:
+        """
+        Process the passed record batch that comes in on the specified anchor.
+
+        The method that gets called whenever the plugin receives a record batch on an input.
+
+        This method IS NOT called during update-only mode.
+
+        Parameters
+        ----------
+        batch
+            A pyarrow Table containing the received batch.
+        anchor
+            A namedtuple('Anchor', ['name', 'connection_name']) containing input connection identifiers.
```

Comment:
TW review:

A namedtuple('Anchor', ['name', 'connection_name']) that contains input connection identifiers.